test/functional/tests/conftest.py

```diff
@@ -123,6 +140,65 @@ def pytest_runtest_setup(item):
     TestRun.LOGGER.start_group("Test body")
 
 
+def base_prepare(item):
+    with TestRun.LOGGER.step("Cleanup before test"):
+        TestRun.executor.run("pkill --signal=SIGKILL fsck")
+        Udev.enable()
+        kill_all_io(graceful=False)
+        DeviceMapper.remove_all()
+
+        if installer.check_if_installed():
+            try:
+                InitConfig.create_default_init_config()
+                unmount_cas_devices()
+                casadm.stop_all_caches()
+                casadm.remove_all_detached_cores()
+            except Exception:
+                pass  # TODO: Reboot DUT if test is executed remotely
+
+        remove(str(opencas_drop_in_directory), recursive=True, ignore_errors=True)
+
+        if Drbd.is_installed():
+            __drbd_cleanup()
+
+        lvms = Lvm.discover()
+        if lvms:
+            Lvm.remove_all()
+            LvmConfiguration.remove_filters_from_config()
+
+        if len(TestRun.disks):
+            raids = Raid.discover()
+            for raid in raids:
+                # stop only those RAIDs, which are comprised of test disks
+                test_run_disk_ids = {dev.device_id for dev in TestRun.disks.values()}
+                if filter(lambda dev: dev.device_id in test_run_disk_ids, raid.array_devices):
+                    raid.umount_all_partitions()
+                    raid.remove_partitions()
+                    raid.stop()
```

Comment:
reintroduced problem with raids, raid.unmount() is needed before stop

you basically reverted my recent fix for raid :/